### PR TITLE
feat: add support for archive snapshot type

### PR DESCRIPTION
**Pull request by okuuva**
Fixes https://github.com/vmware-tanzu/velero/issues/7371.

Signed-off-by: Oula Kuuva <oula.kuuva@custobar.com>

### CODE REVIEW

**Comment by tuusberg** [comment]
@okuuva I just opened a draft PR in this repo that solves the same problem and saw yours :) Great job! 
I ran into an issue when I was working on my PR: [Snapshot](https://pkg.go.dev/google.golang.org/api@v0.63.0/compute/v1#Snapshot) type in Google SDK v0.63.0 does not have SnapshotType property. I solved it by upgrading the SDK to the latest version (v0.167.0) and also bumped Go version, which, most probably deserves a separate PR.

**Comment by blackpiglet** [comment]
@okuuva 
Thanks for the contribution.
The PR misses using the added snapshot type to create the snapshot.

Two places may need the snapshot type.
https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/a450b024fbf3ef113f514473d7a2060a9a9edf3b/velero-plugin-for-gcp/volume_snapshotter.go#L302-L306
https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/a450b024fbf3ef113f514473d7a2060a9a9edf3b/velero-plugin-for-gcp/volume_snapshotter.go#L326-L330

**Comment by okuuva** [comment]
> @okuuva I just opened a draft PR in this repo that solves the same problem and saw yours :) Great job!

Thanks!

> I ran into an issue when I was working on my PR: [Snapshot](https://pkg.go.dev/google.golang.org/api@v0.63.0/compute/v1#Snapshot) type in Google SDK v0.63.0 does not have SnapshotType property. I solved it by upgrading the SDK to the latest version (v0.167.0) and also bumped Go version, which, most probably deserves a separate PR.

Hmmh, where did you get the v0.63.0? To me it seems like it's currently sitting at [v0.143.0](https://github.com/okuuva/velero-plugin-for-gcp/blame/b70c48fe26196ab7a5e7da640355d6dacf65e6ec/go.mod#L16) which isn't the latest but does include the SnapshotType property.

I personally try to separate version bumps from functional changes unless they're absolutely necessary. While bumping both the SDK version and Go version are probably a good idea I'd do those in separate PR(s).

**Comment by okuuva** [comment]
> @okuuva Thanks for the contribution. The PR misses using the added snapshot type to create the snapshot.
> 
> Two places may need the snapshot type.
> 
> https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/a450b024fbf3ef113f514473d7a2060a9a9edf3b/velero-plugin-for-gcp/volume_snapshotter.go#L302-L306
> 
> 
> https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/a450b024fbf3ef113f514473d7a2060a9a9edf3b/velero-plugin-for-gcp/volume_snapshotter.go#L326-L330

...well, that was an awkward oversight from my part 😅 Fixed it now, thanks!

**Comment by tuusberg** [comment]
@okuuva nevermind, it seems my fork was outdated, it's 0.143.0 in the main branch and it would work without any modification.

**Review by gokulkrishnaa — APPROVED** [review]

File: changelogs/unreleased/173-okuuva [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## All changes
```
Comment (blackpiglet) [inline changelogs/unreleased/173-okuuva:1]: I think we can delete this change log file from PR.
The plugin's release note is generated manually, not like the Velero release process.

**Comment by blackpiglet** [comment]
@okuuva 
Thanks for your contribution. This PR looks good. Just delete the changelog file, then we can merge this PR.

File: changelogs/unreleased/173-okuuva (hunk 1 above)
Comment (okuuva) [reply changelogs/unreleased/173-okuuva]: Done. First tried to just revert the commit but forgot to add the signoff to the commit message 😓 So I just ended up dropping both the revert commit and the original with rebase.

**Review by blackpiglet — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Comment by gokulkrishnaa** [comment]
@blackpiglet what release version can we expect this fix to be a part of?

**Comment by blackpiglet** [comment]
This plugin will be shipped with Velero v1.14.0; the version should be v1.10.0.

**Pull request merged**